lib/screens/home/home.dart

```diff
@@ -0,0 +1,145 @@
+    setState(() {
+      selectedYear = year;
+      currentBuilder = courseItemBuilder;
+    });
+  }
+
+  selectCourse(Course course) async {
+    print('Selected course: ' + course.name);
+
+    currentList = await getTAs(course);
+    setState(() {
+      selectedCourse = course;
+      currentBuilder = taItemBuilder;
+    });
+  }
+
+  selectTA(TA ta) {
+    print('Selected TA: ' + ta.name);
+    String title = ta.name + ' - ' + selectedCourse.name;
+    Navigator.push(
+        context,
+        MaterialPageRoute(
+            builder: (context) =>
+                TaCoursePage(title, ta.id, selectedCourse.id)));
+  }
+
+  Widget yearItemBuilder(Year year) {
+    return ListItem(
+      title: year.name,
+      onTap: () => selectYear(year),
+    );
+  }
+
+  Widget courseItemBuilder(Course course) {
+    return ListItem(
+      title: course.name,
+      onTap: () => selectCourse(course),
+    );
+  }
+
+  Widget taItemBuilder(TA ta) {
+    return ListItem(
+      title: ta.name,
+      onTap: () => selectTA(ta),
+      leading: CircleAvatar(
+        backgroundColor: Colors.deepPurple[100],
+        radius: 18,
+        child: Opacity(
+          opacity: 0.9,
+          child: Icon(
+            Icons.person,
+            color: Colors.white,
+          ),
+        ),
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        backgroundColor: Colors.white,
+        appBar: AppBar(
+          title: 'Innopolis Feedback',
+          actions: <Widget>[
+            FlatButton.icon(
+              icon: Icon(Icons.person),
+              label: Text('Sign out?'),
+              onPressed: () => AuthService().signOut(),
+            ),
+          ],
+        ),
+        body: Column(
+          children: [
+            Expanded(
+              child: FutureBuilder(
+                future: getYears(),
+                builder: (context, snapshot) {
+                  if (snapshot.hasData) {
+                    years = snapshot.data;
+                    if (selectedYear == null) {
+                      currentList = years;
+                      currentBuilder = yearItemBuilder;
+                    }
+                    return ListView.builder(
+                      padding: const EdgeInsets.all(30),
+                      itemCount: currentList.length,
+                      itemBuilder: (context, index) =>
+                          currentBuilder(currentList[index]),
+                    );
+                  }
+                  if (snapshot.hasError) {
+                    print(snapshot.error);
+                    return Text('Oops! Something went wrong :(');
+                  }
+                  return Loading();
+                },
+              ),
+            ),
+            if (selectedYear != null || selectedCourse != null)
+              RaisedButton(
+                child: Text('Back'),
+                onPressed: goBack,
+              )
```

Comment:
Shouldn't this be an Action button?

Reply:
This is for later
I will remove this button and replace this functionality with leading back icon in an appBar

Reply:
Fixed
Replaces this button with arrow_back button at AppBar

Reply:
Not good. Check new review.